Assets/MixedRealityToolkit/SpatialSound/Scripts/Effects/AudioInfluencerController.cs

```diff
@@ -0,0 +1,297 @@
+        /// </summary>
+        private DateTime lastUpdate = DateTime.MinValue;
+
+        /// <summary>
+        /// The source of the audio.
+        /// </summary>
+        private AudioSource audioSource;
+
+        /// <summary>
+        /// The initial volume level of the audio source.
+        /// </summary>
+        private float initialAudioSourceVolume;
+
+        /// <summary>
+        /// The hits returned by Physics.RaycastAll
+        /// </summary>
+        private RaycastHit[] hits;
+
+        /// <summary>
+        /// The collection of previously applied audio influencers.
+        /// </summary>
+        private List<IAudioInfluencer> previousInfluencers = new List<IAudioInfluencer>();
+
+        /// <summary>
+        /// Potential effects manipulated by an audio influencer and their key
+        /// properties.
+        /// <summary>
+        private AudioLowPassFilter lowPassFilter;
+        private float nativeLowPassCutoffFrequency;
+        private AudioHighPassFilter highPassFilter;
+        private float nativeHighPassCutoffFrequency;
+
+        private void Awake()
+        {
+            audioSource = gameObject.GetComponent<AudioSource>();
+            initialAudioSourceVolume = audioSource.volume;
+
+            // Get optional filters (and initial values) that the sound designer / developer 
+            // may have applied to this game object
+            lowPassFilter = gameObject.GetComponent<AudioLowPassFilter>();
+            nativeLowPassCutoffFrequency = (lowPassFilter != null) ? lowPassFilter.cutoffFrequency : NeutralHighFrequency;
+            highPassFilter = gameObject.GetComponent<AudioHighPassFilter>();
+            nativeHighPassCutoffFrequency = (highPassFilter != null) ? highPassFilter.cutoffFrequency : NeutralLowFrequency;
+
+            // Preallocate the array that will be used to collect RaycastHit structures.
+            hits = new RaycastHit[MaxObjects];
+        }
+
+        private void Update()
+        {
+            DateTime now = DateTime.Now;
+
+            // Audio influences are not updated every frame.
+            if ((UpdateInterval * 1000.0f) <= (now - lastUpdate).Milliseconds)
+            {
+                audioSource.volume = initialAudioSourceVolume;
+
+                // Get the audio influencers that should apply to the audio source.
+                List<IAudioInfluencer> influencers = GetInfluencers();
+                foreach (IAudioInfluencer influencer in influencers)
+                {
+                    // Apply the influencer's effect.
+                    influencer.ApplyEffect(gameObject);
+                }
+
+                // Find and remove the audio influencers that are to be removed from the audio source.
+                List<IAudioInfluencer> influencersToRemove = new List<IAudioInfluencer>();
+                foreach (IAudioInfluencer prev in previousInfluencers)
+                {
+                    MonoBehaviour mbPrev = prev as MonoBehaviour;
```

Comment:
still not sure if this is correct.  What about classes that implement the IAudioInfluencer that don't inherit from MonoBehaviour?

Reply:
then prev will be null and the following would handle it appropriately.
                        ((mbPrev != null) && !mbPrev.isActiveAndEnabled))
If the IAudioInfluencer does not inherit from MonoBehaviour, it is likely not able to be disabled anyhow.

Reply:
That's what I mean, what if a dev decided to make their own IAudioInfluencer, and didn't want to have their class inherit from Monobehavior?

Reply:
If they make their own IAudioInfluencer that does not inherit from MonoBehavior, they will not be able to attach it to their game objects, which would make the system not function at all for them. It is not designed for use without the influencer being a MonoBehavior.

Reply:
Fair enough